jiant/proj/simple/runscript.py

```diff
@@ -0,0 +1,358 @@
+    --train_task_name_list {",".join(args.train_tasks)} \\
+    --val_task_name_list {",".join(args.val_tasks)} \\
+    --test_task_name_list {",".join(args.test_tasks)} \\
+    --train_batch_size {args.train_batch_size} \\
+    --eval_batch_multiplier 2 \\
+    --epochs {args.num_train_epochs} \\
+    --num_gpus {torch.cuda.device_count()}
+""".strip()
+    if args.train_examples_cap:
+        s += f" \\\n    --train_examples_cap {args.train_examples_cap}"
+    print(s.strip())
+
+    print("\n# === Step 5: Train/Eval! === #")
+    if args.model_path:
+        model_load_mode = "partial"
+        model_path = args.model_path
+    else:
+        # From Transformers
+        if any(task_name.startswith("mlm_") for task_name in full_task_name_list):
+            model_load_mode = "from_transformers_with_mlm"
+        else:
+            model_load_mode = "from_transformers"
+        model_path = os.path.join(
+            args.exp_dir, "models", args.model_type, "model", f"{args.model_type}.p"
+        )
+    s = f"""
+python jiant/proj/main/runscript.py \\
+    run \\
+    --jiant_task_container_config_path \
+{os.path.join(args.exp_dir, "run_configs", f"{args.run_name}_config.json")} \\
+    --output_dir {os.path.join(args.exp_dir, "runs", args.run_name)} \\
+    --model_type {args.model_type} \\
+    --model_path {model_path} \\
+    --model_config_path \
+    {os.path.join(args.exp_dir, "models", args.model_type, "model", f"{args.model_type}.json")} \\
+    --model_tokenizer_path {os.path.join(args.exp_dir, "models", args.model_type, "tokenizer")} \\
+    --model_load_mode {model_load_mode}
+""".strip()
+    if args.train_tasks:
+        s += " \\\n    --do_train"
+    if args.val_tasks:
+        s += " \\\n    --do_val"
+    covered_attrs = [
+        "jiant_task_container_config_path",
+        "output_dir",
+        "model_type",
+        "model_path",
+        "model_config_path",
+        "model_tokenizer_path",
+        "model_load_mode",
+    ]
+    for attr in runscript.RunConfiguration.__attrs_attrs__:
+        if attr.name in covered_attrs:
+            continue
+        if not hasattr(args, attr.name):
+            continue
+        args_attr = getattr(args, attr.name)
+        if attr.default == args_attr:
+            continue
+        if attr.default is None and args_attr is None:
+            continue
+        if (
+            "argparse_kwargs" in attr.metadata
+            and "action" in attr.metadata["argparse_kwargs"]
+            and attr.metadata["argparse_kwargs"]["action"] == "store_true"
+        ):
+            s += f" \\\n    --{attr.name}"
+        else:
+            s += f" \\\n    --{attr.name} {args_attr}"
+    print(s.strip())
+
+
+def main():
+    mode, cl_args = zconf.get_mode_and_cl_args()
+    args = RunConfiguration.default_run_cli(cl_args=cl_args)
+    if mode == "run":
+        run_simple(args)
+    elif mode == "dry_run":
+        dry_run(args)
```

Comment:
I am worried the approach of having `dry_run` and `run` will diverge since there is nothing checking that `dry_run` prints out the correct commands. However, I don't think `eval()` is a valid solution due to security concerns so this approach is fine for now.

Reply:
Agreed, while an automated solution would be preferable, I don't think it's worth the work at this point.